kerastuner/tuners/bayesian.py

```diff
@@ -0,0 +1,285 @@
+import numpy as np
+import random
+import json
+from scipy import optimize as scipy_optimize
+from sklearn import gaussian_process
+
+from ..abstractions.tensorflow import TENSORFLOW_UTILS as tf_utils
+from ..engine import tuner as tuner_module
+from ..engine import oracle as oracle_module
+from ..engine import hyperparameters as hp_module
+
+
+class BayesianOptimizationOracle(oracle_module.Oracle):
+    """Bayesian optimization oracle.
+
+    It uses Bayesian optimization with a underlying Gaussian process model.
+    The acquisition function used is upper confidence bound (UCB).
```

Comment:
Is there is a paper reference for this, it would be good to put the link in the docstring